sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1432,6 +1436,13 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                     {
                         partitionIds = await ListPartitionIdsAsync(connection, cancellationToken).ConfigureAwait(false);
                     }
+                    catch (Exception ex) when ((cancellationToken.IsCancellationRequested)
+                        && (((ex is EventHubsException ehEx) && (ehEx.Reason == EventHubsException.FailureReason.ClientClosed)) || (ex is ObjectDisposedException)))
```

Comment:
What we're doing here is guarding against an expected exception during `Stop` because we're now force-closing the AMQP transport types to avoid waiting for the timeout.   We don't want the exception surfaced to the handler nor does it make sense to keep going and let `PerformLoadBalancingAsync` check the token and throw (which is what would happen) since we already know that it's cancelled.

Reply:
Makes sense - but I wasn't clear on why we couldn't do this:
```suggestion
                    catch (Exception ex) when cancellationToken.IsCancellationRequested
```

Reply:
My initial thought was that we don't want to hide other exceptions from the handler;  we want the other catch block to intercept and surface the exception.   I'm wondering if that's actually the case now that I think about it more; maybe we don't care about anything related to querying partitions since we're not going to keep processing.

Yeah, I think you're right.  That makes more sense.